### PR TITLE
Add CC-BY-SA License.

### LINKED ISSUE #22: No license for example code
I could not find any statement about what license the code examples have. Searching this github repo for the word "license" gave 0 results.

The lack of a license means that the authors reserve all rights by default, and it's illegal to copy or use the code in any way, except as allowed by the law, such as for fair use purposes, parody, etc.

This makes it much harder to use the example code as a starting point for a program, which seems to be the intended use. Since I can't legally copy-paste and modify, I'll instead have to learn from the code and then apply what I've learned in my own code. This might be a good thing for me in the end, because I will learn more and better that way, but it is probably asking too much from the intended users of these docs, and also invites unwitting copyright violations.

So, please include a visible statement about which license the examples are released under, or at least add a file called LICENSE, COPYING or similar to the repo with the license text.

### PR DESCRIPTION
**Pull request by Mariatta**
Text copied from https://choosealicense.com/licenses/cc-by-sa-4.0/

Closes https://github.com/asyncio-docs/asyncio-doc/issues/22

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -52,3 +52,9 @@ See also
 * https://github.com/python/asyncio
 * http://krondo.com/an-introduction-to-asynchronous-programming-and-twisted/
 * https://curio.readthedocs.io/en/latest/tutorial.html
+
+License
+=======
+
+The content of this site is licensed under the `Creative Commons Attribution
+Share Alike 4.0 License <https://creativecommons.org/licenses/by-sa/4.0/>`_ (CC-BY-SA).
```
Comment (vstinner) [inline README.rst]: Can you please also put the same change at the bottom of index.rst?

**Comment by vstinner** [comment]
As I wrote in the issue #22, I suggest to distribute code examples as CC0.

CC BY-SA for code examples is not convenient :-(

Or maybe we need a single license for everything without the annoying "BY" part of CC BY-SA.

**Comment by Mariatta** [comment]
Thanks. I added a sentence saying that code examples are CC0, but everything else is CC BY-SA.
Will this work?
I kept the main LICENSE file in the repo as CC BY-SA.

**Comment by vstinner** [comment]
I requested the review of the most active contributors to the documentation, since it's a license change.

**Review by vstinner — APPROVED** [review]
LGTM. But it would be nice to wait the feedback from the other contributors to this documentation, since it's a license change.

I'm not sure that it's useful to have a copy of the license in the LICENSE file. If you think that it's useful, would you mind to also add the CC0 license?

**Review by vxgmichel — APPROVED** [review]
I'm ok with it!

**Comment by Mariatta** [comment]
Ok. I added a new file: LICENSE_CC0 which contains the text for CC0 License. :)

**Review by hoh — APPROVED** [review]

**Comment by matrixise** [comment]
Hi @vstinner @Mariatta @hoh and @vxgmichel 

I am going to merge this one if you agree.

**Review by ludovic-gasc — APPROVED** [review]
LGTM

**Pull request merged**